bugrap/src/main/java/org/vaadin/bugrap/ui/columns/FineDateRenderer.java

```diff
@@ -1,64 +1,57 @@
 package org.vaadin.bugrap.ui.columns;
 
-import java.io.ObjectInputStream.GetField;
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.util.Calendar;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
 import java.util.Date;
-import java.util.TimeZone;
 
 import com.vaadin.ui.renderers.DateRenderer;
 
 import elemental.json.JsonValue;
 
 public class FineDateRenderer extends DateRenderer{
 
-	private final int[] units = {Calendar.YEAR, Calendar.MONTH, Calendar.DAY_OF_YEAR, Calendar.HOUR_OF_DAY, Calendar.MINUTE, Calendar.SECOND};
-	private final String[] unitTextValues = {"year", "month", "day", "hour", "minute", "second"};
-	
-	
 	@Override
 	public JsonValue encode(Date value) {
 		return super.encode(getFineTextFromDate(value), String.class);
+		
 	}
 	
-	private String getFineTextFromDate(Date value) {
-		StringBuffer presentationText = new StringBuffer();
-		if (value == null)
-			return presentationText.toString();
+	private static String getFineTextFromDate(Date value) {
+		long durationAmount = 0;
+		String durationUnitTxt = "";
 		
-		long diffInMillis = (new Date()).getTime() - value.getTime();
+		Instant now = Instant.now();
+		Instant from = value.toInstant();
 		
-		Calendar diff = Calendar.getInstance();
-		diff.setTimeInMillis(diffInMillis - diff.get(Calendar.ZONE_OFFSET));
-		
-		int index = findMaxUnitIndex(diff);
-		int timeValue = getTimeValueInIndexUnit(diff, index);
-		
-		presentationText.append(timeValue + " " + unitTextValues[index]) ;
-		if (timeValue > 1)
-			presentationText.append("s");
+		if (ChronoUnit.DAYS.between(from, now) > 0) {
+			Period period = Period.between(LocalDate.ofInstant(from, ZoneOffset.UTC), LocalDate.ofInstant(now, ZoneOffset.UTC));
+			if (period.getYears() > 0) {
+				durationAmount = period.getYears();
+				durationUnitTxt = "year";
+			}else if (period.getMonths() > 0) {
+				durationAmount = period.getMonths();
+				durationUnitTxt = "month";
+			}else if (period.getDays() > 0) {
+				durationAmount = period.getDays();
+				durationUnitTxt = "day";
+			}
+		}else if (ChronoUnit.HOURS.between(from, now) > 0) {
+			durationAmount = ChronoUnit.HOURS.between(from, now);
+			durationUnitTxt = "hour";
+		}else if (ChronoUnit.MINUTES.between(from, now) > 0) {
+			durationAmount = ChronoUnit.MINUTES.between(from, now);
+			durationUnitTxt = "minute";
+		}
```

Comment:
What about seconds?

Reply:
Less than a minute is thought as "now", do you think it should be too precise?
Cos even if it shows 5 secs, it will be more than 5 secs till you read it :)

Reply:
Ok.